### PR TITLE
doc(replays): Add basic docs for Replays

### PR DESCRIPTION
**Pull request by jan-auer**
Contains bare minimum documentation for the `relay-replays` module. More documentation on the payload and processing functionality, similar to `relay-metrics` and `relay-profiling` should be added in a followup.

#skip-changelog

### CODE REVIEW

File: relay-replays/src/recording.rs [hunk 1]
```diff
@@ -109,9 +145,24 @@ impl<'a> ReplayScrubber<'a> {
         }
     }
 
-    /// Parses compressed replay recording payloads and applies data scrubbers.
+    /// Parses a replay recording payloads and applies data scrubbers.
```
Comment (olksdr) [inline relay-replays/src/recording.rs]:
```suggestion
    /// Parses replay recording payloads and applies data scrubbers.
```

**Review by olksdr — APPROVED** [review]
:shipit:

File: relay-replays/src/lib.rs [hunk 2]
```diff
@@ -1,2 +1,17 @@
+//! Session replay protocol and processing for Sentry.
+//!
+//! [Session Replay] provides a video-like reproduction of user interactions on a site or web app.
+//! All user interactions — including page visits, mouse movements, clicks, and scrolls — are
+//! captured, helping developers connect the dots between a known issue and how a user experienced
+//! it in the UI.
+//!
+//! [session replay]: https://docs.sentry.io/product/session-replay/
```
Comment (iker-barriocanal) [inline relay-replays/src/lib.rs:8]: nit, as the case is in the linked docs
```suggestion
//! [Session Replay]: https://docs.sentry.io/product/session-replay/
```

**Review by iker-barriocanal — APPROVED** [review]

File: relay-replays/src/lib.rs (hunk 2 above)
Comment (jan-auer) [reply relay-replays/src/lib.rs:8]: That's intentional, my markdown utilities do that automatically. Links are case insensitive, which is a useful feature to support linking to the same document if you use the same keyword multiple times with different casing (e.g. at the beginning and in the middle of sentences).

Great catch, though :)

**Pull request merged**